oauthenticator/globus.py

```diff
@@ -31,36 +32,6 @@ class GlobusMixin(OAuth2Mixin):
     @return_future
     def authorize_redirect(self, client=None, callback=None):
         self.redirect(client.oauth2_get_authorize_url())
-        callback()
-
-
-class GlobusOAuthCallbackHandler(OAuthCallbackHandler):
-    """This extra callback handler is needed to store transfer tokens
-    in auth_state. This is needed since there isn't a sensible way to
-    access the database through `authenticate`, although this may change
-    with the following issue:
-    https://github.com/jupyterhub/jupyterhub/issues/1063
-    globus tokens are saved in the database to persist server
-    restarts and closed browser windows. This ensures consistency -- whenever
-    the user is logged in, they will be able to spawn a notebook with tokens.
-    """
-    @gen.coroutine
-    def get(self):
-        username = yield self.authenticator.get_authenticated_user(self, None)
-
-        if username:
-            user = self.user_from_username(username)
-            self.set_globus_data(user)
-            self.set_login_cookie(user)
-            self.redirect(url_path_join(self.hub.server.base_url, 'home'))
-        else:
-            raise web.HTTPError(403)
-
-    def set_globus_data(self, user):
-        user.auth_state = {
-            'globus_data': self.authenticator.globus_data
-        }
-        self.db.commit()
```

Comment:
No longer needed, now that passing data to `auth_state` is supported.

Reply:
What is the `auth_state` here? The return of the `authenticate` method?

Reply:
`auth_state` is brand new and mainly the reason for the updates. The Jupyterhub folks didn't like sensitive material being non-encrypted within the Jupyterhub database at rest. They changed `auth_state` from a regular JSON field to an encrypted JSOS field, and data is saved by returning a dict here from the `authenticate` method instead of calling out to the `user` class specifically. 

Oauthenticator just had another update this morning, so this may change: https://github.com/jupyterhub/oauthenticator/issues/110